### PR TITLE
Make main nav collapsible

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Resolves https://github.com/tektoncd/dashboard/issues/803

Make the SideNav component collapsible to provide more screen real
estate for the main content at smaller window sizes.

Also add a 'skip to content' link for keyboard and screen reader users.

Before / After:
image

Collapsed:
image

This PR also contains a fix for a style specificity bug affecting the page header on production builds (local dev builds not affected).

Before:
image

After:
image

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: packages/components/src/components/Header/Header.js [hunk 1]
```diff
@@ -17,31 +17,46 @@ import { injectIntl } from 'react-intl';
...
+          }
+          isCollapsible
+          onClick={onHeaderMenuButtonClick}
         />
-      </span>
-      <HeaderName href="/" prefix="">
-        Tekton
+      ) : null}
+      <HeaderName href="/" prefix="Tekton">
+        Dashboard
```
Comment (steveodonovan) [inline packages/components/src/components/Header/Header.js:59]: I'm guessing dashboard doesn't get translated ?

File: packages/components/src/components/Header/Header.js [hunk 2]
```diff
@@ -17,31 +17,46 @@ import { injectIntl } from 'react-intl';
...
-            id: 'dashboard.logo.tooltip',
-            defaultMessage: 'Meow'
-          })}
+      <SkipToContent>
+        {intl.formatMessage({
+          id: 'dashboard.skipToContent',
+          defaultMessage: 'Skip to main content'
+        })}
+      </SkipToContent>
+      {onHeaderMenuButtonClick ? (
```
Comment (steveodonovan) [inline packages/components/src/components/Header/Header.js:41]: `onHeaderMenuButtonClick && ( ...`

**Review by steveodonovan — COMMENTED** [review]
Few nits again, overall fine

File: packages/components/src/components/Header/Header.js (hunk 1 above)
Comment (AlanGreene) [reply packages/components/src/components/Header/Header.js:59]: yeah, product names generally don't get translated

**Review by steveodonovan — APPROVED** [review]
/approve
/lgtm

**Pull request merged**